### ISSUE TITLE
Possibility to upload non-mandatory (virus) metadata

### ISSUE THREAD

**Issue by tpall** [usegalaxy-eu/ena-upload-cli#9]
Thanks for the nice helper. However, I noticed that currently ena-upload-cli supports only mandatory [ERC000033](https://www.ebi.ac.uk/ena/browser/view/ERC000033) fields in their XML templates. I came up for myself with a little hack with updated XML forms  https://github.com/avilab/ena-upload-cli/tree/location to get some additional metadata uploaded, e.g. age, geographic location locality+lon/lat. I appreciate that under current implementation there is no simple fix to include any of optional checklist fields, given that ENA database may not accept empty fields(not sure).

One way to fix this possible issue would be, very briefly, to check imported tables against schemas and serialise dictionaries to XML. So that any combination of non-mandatory fields can be included.

I happy to be corrected and directed to the right path if there is a way to include optional/recommended (virus) metadata fields using this app as it is.

**Comment by bgruening** [comment]
Thanks a lot, @tpall for looking into it. Could you set the entire `SAMPLE_ATTRIBUTE` in an if condition, so that it is only set when the variable is also available?

Something like that?

```
<py:if test="row.geographic_location_region">
            <SAMPLE_ATTRIBUTE>
                <TAG>geographic location (region and locality)</TAG>
                <VALUE>${row.geographic_location_region}</VALUE>
            </SAMPLE_ATTRIBUTE>
</py:if>
```

**Comment by bedroesb** [comment]
this was solved in #10
